python-sdk/src/transfers/data_providers/aws/s3.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import annotations
+
+import logging
+import os
+
+from airflow.hooks.dbapi import DbApiHook
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from transfers.constants import FileLocation, TransferMode
+from transfers.data_providers import create_dataprovider
+from transfers.data_providers.base import DataProviders
+from transfers.datasets.base import UniversalDataset as Dataset
+
+from astro.constants import LoadExistStrategy
+
+# from transfers.utils import get_dataset_connection_type
+
+
+class S3DataProvider(DataProviders):
+    """
+    DataProviders interactions with S3 Dataset.
+    """
+
+    def __init__(
+        self,
+        conn_id: str,
+        extra: dict = {},
+        transfer_params: dict = {},
+        transfer_mode: TransferMode = TransferMode.NONNATIVE,
+        if_exists: LoadExistStrategy = "replace",
+    ):
+        super().__init__(
+            conn_id=conn_id,
+            extra=extra,
+            transfer_params=transfer_params,
+            transfer_mode=transfer_mode,
+            if_exists=if_exists,
+        )
+        self.transfer_mapping: set = {
+            FileLocation.AWS,
+            FileLocation.google_cloud_platform,
+        }
+        self.LOAD_DATA_FROM_SOURCE = {
+            "google_cloud_platform": "load_data_from_gcs",
+            "aws": "load_data_from_s3",
+        }
+
+    @property
+    def hook(self) -> DbApiHook:
+        """Return an instance of the database-specific Airflow hook."""
+        return S3Hook(
+            aws_conn_id=self.conn_id,
+            verify=self.extra.get("verify", None),
+            transfer_config_args=self.extra.get("transfer_config_args", None),
+            extra_args=self.extra.get("s3_extra_args", {}),
+        )
+
+    def check_if_exists(self, dataset: Dataset) -> bool:
+        """Return true if the dataset exists"""
+        raise NotImplementedError
+
+    def load_data_from_gcs(self, source_dataset: Dataset, destination_dataset: Dataset) -> None:
```

Comment:
Wait are we going to have a function for each ingestor? This seems like it could blow up the classes. Unless this is more meant for "special cases"?